File: templates/problems/05_binary_search/07_median_two_sorted/go/median_two_sorted_test.go
package median_two_sorted

import (
	"math"
	"testing"
)

func TestFindMedianSortedArrays(t *testing.T) {
	tests := []struct {
		name  string
		nums1 []int
		nums2 []int
		want  float64
	}{
		{"odd total", []int{1, 3}, []int{2}, 2.0},
		{"even total", []int{1, 2}, []int{3, 4}, 2.5},
		{"first empty", []int{}, []int{1}, 1.0},
		{"second empty", []int{2}, []int{}, 2.0},
		{"no overlap", []int{1, 2}, []int{3, 4, 5}, 3.0},
		{"interleaved", []int{1, 3, 5}, []int{2, 4, 6}, 3.5},
		{"single elements", []int{1}, []int{2}, 1.5},
		{"duplicates", []int{1, 1, 1}, []int{1, 1, 1}, 1.0},
	}
	for _, tt := range tests {
		t.Run(tt.name, func(t *testing.T) {
			got := findMedianSortedArrays(tt.nums1, tt.nums2)
			if math.Abs(got-tt.want) > 1e-5 {
				t.Errorf("findMedianSortedArrays(%v, %v) = %f, want %f", tt.nums1, tt.nums2, got, tt.want)
			}
		})
	}
}

File: templates/problems/05_binary_search/07_median_two_sorted/typescript/median_two_sorted.test.ts
import { describe, it, expect } from "vitest";
import { findMedianSortedArrays } from "./median_two_sorted";

describe("findMedianSortedArrays", () => {
  it("odd total", () => {
    expect(findMedianSortedArrays([1, 3], [2])).toBe(2.0);
  });

  it("even total", () => {
    expect(findMedianSortedArrays([1, 2], [3, 4])).toBe(2.5);
  });

  it("first empty", () => {
    expect(findMedianSortedArrays([], [1])).toBe(1.0);
  });

  it("second empty", () => {
    expect(findMedianSortedArrays([2], [])).toBe(2.0);
  });

  it("no overlap", () => {
    expect(findMedianSortedArrays([1, 2], [3, 4, 5])).toBe(3.0);
  });

  it("interleaved", () => {
    expect(findMedianSortedArrays([1, 3, 5], [2, 4, 6])).toBe(3.5);
  });

  it("single elements", () => {
    expect(findMedianSortedArrays([1], [2])).toBe(1.5);
  });

  it("duplicates", () => {
    expect(findMedianSortedArrays([1, 1, 1], [1, 1, 1])).toBe(1.0);
  });
});
